### ISSUE TITLE
NET-3165 Fix S3971 FP: Support IAsyncLifetime (xUnit v3)

### ISSUE THREAD

**Issue by dnnr** [SonarSource/sonar-dotnet#9777]
### Description

Rule S3971 raises a false positive when calling `GC.SuppressFinalize` in a `DisposeAsync` method of a class that implements [`IAsyncLifetime` from xUnit](https://github.com/xunit/xunit/blob/5607c6bd70892605015aeaeca1d330f4a746f71f/src/xunit.v3.core/IAsyncLifetime.cs). The interface itself inherits `IAsyncDisposable`, so S3971 should accept this accordingly.

Unlike what I initially assumed, this doesn't seem to be a regression of #3639, but is caused by the intermediate interface. In the attached reproducer, `Foo1` is the only case where S3971 is triggered.

### Reproducer

```
class Foo1 : IAsyncLifetime
{
    public ValueTask DisposeAsync()
    {
        GC.SuppressFinalize(this); // <- S3971 complains here
        return ValueTask.CompletedTask;
    }

    public ValueTask InitializeAsync() => throw new NotImplementedException();
}

class Foo2 : IAsyncDisposable
{
    public ValueTask DisposeAsync()
    {
        GC.SuppressFinalize(this);
        return ValueTask.CompletedTask;
    }
}

class Foo3 : IDisposable
{
    public void Dispose()
    {
        GC.SuppressFinalize(this);
    }
}
```

### Product and Version

SonarQube Server v2025.3.1

**Comment by pavel-mikula-sonarsource** [comment]
Hi @dnnr,

Thank you for reporting this case. I confirm it as False Positive.

**Comment by pavel-mikula-sonarsource** [comment]
@dnnr, I have issue with reproducing the issue. I can't get it raised with the provided example.

What's the target framework of your project, exact xunit package(s) that you reference and their versions?

**Comment by dnnr** [comment]
My target framework is net8.0, and going by the packages lock file history, xUnit has been consistently at [v3.2.0](https://www.nuget.org/packages/xunit.v3/3.2.0) since I've reported this.

Unfortunately though, I cannot reproduce the FP anymore right now either. It might be due to the fact that the instance available to me has since been upgraded to v2025.6.1, but that's nothing more than a guess.

I can understand if that means you'll have to close to issue for now. I can always reopen in case the problem resurfaces.

**Comment by pavel-mikula-sonarsource** [comment]
I don't see a change in that rule between those two versions, so I'm not sure why this doesn't reproduce anymore.

Let's close this for now then. If you face the same issue again, feel free to reopen it.
